pkg/types/vsphere/validation/featuregates.go

```diff
@@ -58,6 +59,11 @@ func GatedFeatures(c *types.InstallConfig) []featuregates.GatedInstallConfigFeat
 			Condition:       hasDataDisks(computeDefs), // Here we need to check disk count
 			Field:           field.NewPath("compute", "platform", "vsphere", "dataDisks"),
 		},
+		{
+			FeatureGateName: features.FeatureGateOnPremDNSRecords,
+			Condition:       v.DNSRecordsType == configv1.DNSRecordsTypeExternal,
+			Field:           field.NewPath("platform", "vsphere", "DNSRecordsType"),
```

Comment:
```suggestion
			Field:           field.NewPath("platform", "vsphere", "dnsRecordsType"),
```

nit: we should use the serialized name of the field for field path, right?